pixl_dcmd/src/pixl_dcmd/_database.py

```diff
@@ -128,27 +128,30 @@ def get_unexported_image(
     MRN + accession number. If this fails as well, raise a NoResultFound.
     """
     try:
-        existing_image: Image = (
-            pixl_session.query(Image)
-            .join(Extract)
-            .filter(
-                Extract.slug == project_slug,
-                Image.study_uid == study_info.study_uid,
-                Image.exported_at == None,  # noqa: E711
-            )
-            .one()
+        existing_image = _query_and_raise_if_exported(
+            pixl_session,
+            [Extract.slug == project_slug, Image.study_uid == study_info.study_uid],
         )
     # If no image is found by study UID, try MRN + accession number
     except exc.NoResultFound:
-        existing_image = (
-            pixl_session.query(Image)
-            .join(Extract)
-            .filter(
+        existing_image = _query_and_raise_if_exported(
+            pixl_session,
+            [
                 Extract.slug == project_slug,
                 Image.mrn == study_info.mrn,
                 Image.accession_number == study_info.accession_number,
-                Image.exported_at == None,  # noqa: E711
-            )
-            .one()
+            ],
         )
     return existing_image
+
+
+def _query_and_raise_if_exported(
+    pixl_session: Session, clause_list: list[ColumnElement[bool]]
+) -> Image:
+    existing_image: Image = (
+        pixl_session.query(Image).join(Extract).filter(*clause_list).one()
+    )
+    if existing_image.exported_at is not None:
+        msg = "Study already exported"
+        raise exc.NoResultFound(msg)
```

Comment:
I think if this exception is raised by the first call to `query_and_raise_if_exported`, the exception would be ignored and the query made again using MRN and accession number. Perhaps this could raise a different exception type to avoid querying the same exported image?

Reply:
Ah yeah that's not a bad point. I guess if we have a study uid and if its been exported then you're right, we shouldn't try again